proposals/p1964.md

```diff
@@ -0,0 +1,140 @@
+# Character literals
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1964)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+    -   [Types](#types)
+    -   [Encoding](#encoding)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Problem
+
+This proposal specifies lexical rules for constant characters in Carbon.
```

Comment:
This is a problem that this proposal addresses, but we are looking for something a bit different in this section: what is the problem a Carbon developer would have that motivates us to add this feature to the language?

You might motivate it with use cases like this code:

```
if (c >= 'A' and c <= 'Z') {
  c += 'a' - 'A';
}
```

This example makes it clear that we are going to treat these differently than strings: they are represented using integers rather than buffers, and have different operations as well.

Reply:
Does this sound good?

The advantage of having an explicit character type fundimentally comes down to characters being represented as integers whereas strings are represented as buffers. This will allow characters to have differenct operations, and be more familar to use. For example:

```
if (c >= 'A' and c <= 'Z') {
  c += 'a' - 'A';
}
```

The example above shows how we would be able to use operations similar to integers. Being able to use the comparison operations and supporting arithmetic operations provides intuitve approach to using characters. This allows us to remove unnecessary logic of type conversion and other control flow logic, that is needed to work with a single element string.